### PR TITLE
Add invalid status/check when trying to build non-master

### LINKED ISSUE #246: github-push - create an invalid status/check when building non-master
## Expected Behaviour

We should update the status / checks API when building from non-master, since there is currently no feedback given.

## Current Behaviour

We create an audit message but don't provide feedback to the user via checks API / status.

```
[github-push] alexellis/not-fn: 'refusing to build non-master branch: refs/heads/faas-team'
```

## Steps to Reproduce (for bugs)

1. Add GitHub App to new repo
2. In a new branch add stack.yml and functions
3. Do a push

### PR DESCRIPTION
**Pull request by ivanayov**
Non-master branches are not built by design. By the moment we used to send only audit message, but never provided feedback to the user. With this change status failure is going to be reported

Signed-off-by: Ivana Yovcheva <iyovcheva@vmware.com>

Closes #246

## How Has This Been Tested?

Tested by @martindekov with `docwareiy/github-push:non-master`

Pushing on non-master branch resulted in
```
stack-deploy
 Failure 
 ran 3 minutes ago in less than 10 seconds

 541a1ef by @martindekov

 non-master

Deploy to OpenFaaS
refusing to build non-master branch: refs/heads/non-master
```

## How are existing users impacted? What migration steps/scripts do we need?
N/A

## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required) N/A
- [x] read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] signed-off my commits with `git commit -s`
- [ ] added unit tests N/A

### CODE REVIEW

**Comment by alexellis** [comment]
Please do not submit PRs which haven’t been tested since it blocks others from completing the work and you’ve given no schedule on when you plan to test the work. 

In this instance maybe you can mark the PR as [WIP] ?

**Comment by martindekov** [comment]
@ivanayov can you please push an image with the change in dockerhub so I can directly deploy the function and test it out?

**Comment by ivanayov** [comment]
@martindekov you can use `docwareiy/github-push:non-master`

**Comment by martindekov** [comment]
Deployed your image `docwareiy/github-push:non-master` and pushed to literally named `non-master` branch which is non master the commit status in github is the following:

### Deploy to OpenFaaS

refusing to build non-master branch: refs/heads/non-master

**Comment by ivanayov** [comment]
Great, thanks @martindekov !

File: github-push/handler.go [hunk 1]
```diff
@@ -100,14 +103,14 @@ func Handle(req []byte) string {
...
 		audit.Post(auditEvent)
+
+		status.AddStatus(sdk.StatusFailure, msg, sdk.StackContext)
+		reportStatus(status)
 		return msg
 	}
 
 	serviceValue := sdk.FormatServiceName(pushEvent.Repository.Owner.Login, pushEvent.Repository.Name)
 
-	eventInfo := sdk.BuildEventFromPushEvent(pushEvent)
```
Comment (alexellis) [inline github-push/handler.go]: Why was this removed?

Comment (alexellis) [reply github-push/handler.go]: OK it was moved up. Can you confirm that the original path still works fine? i.e. "pending status" is sent?

Comment (ivanayov) [reply github-push/handler.go]: It's moved up before the check as we need the `status` object

Comment (martindekov) [reply github-push/handler.go]: I will test the change and post comment underneath

**Comment by martindekov** [comment]
Again test non master:
```
stack-deploy
 Failure 
 ran just now in less than 5 seconds

 25c4c8c by @martindekov

 non-master

Deploy to OpenFaaS
refusing to build non-master branch: refs/heads/non-master
```
On master:
```
gocloudfunc
 Success

 ran 2 minutes ago in half a minute

 def1630 by @martindekov

 master

Build gocloudfunc
deployed: martindekov-gocloudfunc
```
and
```
stack-deploy
 Success

 ran 2 minutes ago in 7 minutes

 def1630 by @martindekov

 master

Deploy to OpenFaaS
stack is successfully deployed

View more details on OpenFaasCloudTest
```

**Comment by martindekov** [comment]
Found a bug while testing that status stays on pending even though `stack.yml` file is in invalid form. Related to https://github.com/openfaas/openfaas-cloud/commit/aa5828702f0e82f5b1070cb1c1b73ff908ceb4e3

**Comment by ivanayov** [comment]
It's not reporting failing status when there is an error in reading stack.yml.
I'll add the status update. 

This is not related to this PR, but to #281 instead. @martindekov was testing with the git-tar image from that PR and without the related github-push image, which reproduced empty raw url.
